lib/Throttle.php

```diff
@@ -109,17 +110,56 @@ public function applyBruteForcePolicy($uid, $ip) {
 	 * @param string $ip
 	 * @return void
 	 */
-	public function clearSuspiciousAttemptsForUidIpCombination($uid, $ip) {
-		$this->attemptMapper->deleteSuspiciousAttemptsForUidIpCombination($uid, $ip);
+	public function clearFailedLoginAttemptsForUidIpCombination($uid, $ip) {
+		$this->loginAttemptMapper->deleteFailedLoginAttemptsForUidIpCombination($uid, $ip);
+	}
+
+	/**
+	 * @param string $token
+	 * @param string $ip
+	 * @return void
+	 */
+	public function addFailedLinkAccess($token, $ip) {
+		$access = new FailedLinkAccess();
+		$access->setLinkToken($token);
+		$access->setIp($ip);
+		$access->setAttemptedAt($this->timeFactory->getTime());
+		$this->linkAccessMapper->insert($access);
+	}
+
+	/**
+	 * @param string $token
+	 * @param string $ip
+	 * @throws LinkAuthException
+	 */
+	public function applyBruteForcePolicyForLinkShare($token, $ip) {
+		$banPeriod = $this->config->getBruteForceProtectionBanPeriod();
+		$banUntil = $this->linkAccessMapper->getLastFailedAccessTimeForTokenIpCombination($token, $ip) + $banPeriod;
+		if ($this->linkAccessMapper->getFailedAccessCountForTokenIpCombination($token, $ip) >=
+			$this->config->getBruteForceProtectionFailTolerance() &&
+			$banUntil > $this->timeFactory->getTime()) {
+			throw new LinkAuthException($this->l->t("Too many failed attempts. Try again in %s.",
```

Comment:
It's questionable to have translated messages in the exception.
* If the exception isn't caught this likely means that any other exception that could happen around here won't be caught neither, with the risk of exposing unneeded info
* If it's caught, it should be the responsibility of whoever caught the exception (probably a controller) to provide a proper response with the translated message.

Reply:
@jvillafanez I agree with your point, but I do not think we can easily implement consistent exception catching mechanism since an event can be dispatched from multiple places and multiple endpoints.
I followed the same way with login protection here. As far as I know, we have some translated exceptions like `HintException` in the codebase also. I would suggest keeping this one in that way for now to reduce effort.

Reply:
ok, you might want to include a comment instead so we know the text will be shown to the users